### ISSUE TITLE
Drop-down lists always open on the primary monitor

### ISSUE THREAD

**Issue by PimDoos** [obsproject/obs-studio#5565]
### Operating System Info

Windows 10

### OBS Studio Version

27.1.3

### OBS Studio Log URL

https://obsproject.com/logs/Tgzt35vXsAKd30v8

### Expected Behavior

When clicking a drop-down list, it's contents are displayed under the button.

### Current Behavior

When clicking a drop-down list, it's contents are displayed on the primary monitor instead of the monitor running the OBS window.

### Steps to Reproduce

1. Open OBS
2. Open settings
3. Move the settings window to a non-primary monitor
4. Click the 'language' drop-down list
5. The list opens on the wrong display

### Anything else we should know?

Screenshot: Primary display on the left, secondary display on the right
![image](https://user-images.githubusercontent.com/17709721/143124919-b4759422-ac2e-4872-8221-a48374d4fbbf.png)

**Comment by notr1ch** [comment]
Are you using any 3rd party display management / window management software?

**Comment by PimDoos** [comment]
No, I'm not using any 3rd party display management.

**Comment by Azeirah** [comment]
I'm not 100% sure if this is caused by the same steps I found, but if you have application scaling in Windows Display settings set to two different numbers

(ie, left screen scaling = 125%, right screen scaling = 100%)

Then the dropdown also gets offset. There are various bugs like this in Qt, see [this one](https://bugreports.qt.io/browse/QTBUG-52606)and [this one](https://bugreports.qt.io/plugins/servlet/mobile#issue/QTBUG-84462) for instance.

Note, my monitors look strange because it is (1) a cropped image and (2) my left monitor is portrait mode and right one is landscape.

![afbeelding](https://user-images.githubusercontent.com/4040870/143302408-632ca95e-76f9-4787-a1f4-bc047954838c.png)

**Comment by Azeirah** [comment]
> No, I'm not using any 3rd party display/window management.

Are both your monitors set to the same resolution?

Secondly, do both your monitors have the same value for application scaling? (right click on desktop > display settings > check the scale and layout value for monitor 1 and for monitor 2)

**Comment by PimDoos** [comment]
My monitor setup is 3 x 1920x1080 / 60 Hz @ 100% scaling. 
Your issue looks similar, but also has different behaviour. For me, the drop-down list is displayed at the same coordinates, but on the wrong monitor. On yours, it looks like the positioning is also off (probably the same coordinates * scaling factor)

**Comment by jp9000** [comment]
Wow that is bizarre.

**Comment by WizardCM** [comment]
I have managed to reproduce this on my work PC. I'll do further tests, as there's nothing special about its configuration - two 1080p 60Hz displays at 100% scaling running on Windows 10 21H2. However, the two displays are connected to different GPUs.

As far as I can tell, the first attempt at opening a dropdown (either in the main window, Properties, or Settings), it'll spawn on the wrong display. Every subsequent open displays the dropdown in question on the wrong display until it's fully open, then it teleports to the correct display. This behaviour occurs fresh on each new dropdown opened.

![obs-dropdown](https://user-images.githubusercontent.com/941350/146905609-23cdb216-f038-49d6-85ac-dd83c49ae9d0.png)

**Comment by PimDoos** [comment]
Good call on the GPUs. My displays are on different GPUs as well.

**Comment by WizardCM** [comment]
Alright, new test with 3 displays - ordered 1, 2, 3, starting with 1 being the main display and 3 being on the iGPU. OBS is running on the AMD Radeon HD 6350.

* If display 1 or 2 is the main display, dropdowns **always teleport** to the "main" display, regardless of whether the main display 1, 2 or 3.

* If display 3 (iGPU) is the main display, then the dropdown **always** spawns in the correct location.

* Unfortunately, I am unable to launch OBS on the iGPU to determine whether it changes the behaviour.

**Comment by nwb99** [comment]
I'm new to OBS and stumbled across this today (Jan 26, 2022).
I have an older RX 580 (GPU prices!), so I figured I'd hook my secondary monitor to my motherboard and use the iGPU on the 10700k. Drop down menus and tooltips appear on the primary monitor rather than the monitor in which the OBS window is displayed. I swapped secondary monitor back to be on the AMD GPU like before, and it behaves as expected.

The issue does not occur when OBS is on a display that is attached to the same GPU as the primary display.

2x 1920x1080 monitors at 100% scaling in Windows 10, build 19043.1466. AMD RX 580 and Intel UHD 630 (i7-10700k)
